cache/edge_lambdas/src/redirector.test.ts

```diff
@@ -72,50 +72,70 @@ jest.mock('./redirects', () => {
     ...defaultRedirects,
     queryRedirects: {
       ...defaultRedirects.queryRedirects,
-      '/test': {
-        matchParams: new URLSearchParams({
-          beep: 'boop',
-          bing: 'bong',
-          fizz: 'buzz',
-        }),
-        forwardParams: new Set(),
-        redirectPath: '/test-destination',
-      },
+      '/test': [
+        {
+          matchParams: new URLSearchParams({
+            beep: 'boop',
+            bing: 'bong',
+            fizz: 'buzz',
+          }),
+          forwardParams: new Set(),
+          redirectPath: '/test-destination',
+        },
+      ],
     },
   };
 });
 
 describe('Query string redirects', () => {
-  test('Occur when all the params in the definition are in the request', () => {
-    const redirectedResponse = getRedirect(
-      request({ uri: '/works', querystring: 'search=images' })
-    );
-
-    expect(redirectedResponse?.status).toEqual('301');
-    expect(redirectedResponse?.headers.location[0]).toEqual({
-      key: 'Location',
-      value: 'https://wellcomecollection.org/images',
-    });
-  });
-
-  test('Do not occur if all of the params in the definition are not matched', () => {
-    const nonRedirectedResponse = getRedirect(
-      request({ uri: '/test', querystring: 'bing=bong' })
-    );
-    expect(nonRedirectedResponse).toBeUndefined();
-  });
+  test.each([
+    // Offically active cases
+    // listed in redirect.ts rules
+    // as we can't have them work locally, we test them here
+    {
+      testFunction:
+        'Image search within works should redirect to Image search hub page',
+      from: { uri: '/works', querystring: 'search=images' },
+      to: 'https://wellcomecollection.org/search/images',
+    },
+    {
+      testFunction:
+        'Works should redirect to Catalogue search hub page, with specified forwardParams§',
+      from: { uri: '/works', querystring: 'query=beep&workType=v&beep=boop' },
+      to: 'https://wellcomecollection.org/search/works?query=beep&workType=v',
+    },
+    {
+      testFunction:
+        'Image should redirect to Image search hub page, with specified forwardParams§',
+      from: { uri: '/images', querystring: 'images.color=blue&hello=world' },
+      to: 'https://wellcomecollection.org/search/images?images.color=blue',
+    },
+    // Extra testing cases
+    {
+      testFunction: 'Only forward the specified forwardParams',
+      from: { uri: '/images', querystring: 'beep=boop' },
+      to: 'https://wellcomecollection.org/search/images',
+    },
+    {
+      testFunction: 'Do not occur if the uri is not an exact match',
+      from: { uri: '/work' },
+      to: 'redirectionShouldFail',
+    },
+    {
+      testFunction: 'Do not occur if the uri is not an exact match',
+      from: { uri: '/test', querystring: 'query=beep' },
```

Comment:
This is good - my 2 comments are around making the test cases simpler and moving some logic out of the head of the person writing/reading the tests and into the test body, so we could write test cases like this:
```typescript
test.each([
  {
     // Nit: changed from `testFunction` to `description` as it's not a function
     description: 'Works should redirect to Catalogue search hub page, with specified forwardParams'
     from: '/works?query=beep&workType=v&beep=boop',
     to: '/search/works?query=beep&workType=v'
  },
  // ...
])('$description', ({ from, to }) => {
  // ..
});
```

(1) If we change these test cases to be of type `{ from: string }` and then have something in the body of the test like this, we don't have the (arbitrary) split-up URLs in the test cases.

```typescript
const fromUrl = new URL(from);
fromUrl.searchParams.sort(); // strictly, we're not testing whether it maintains query string order 
const fromRequestObject = {
  uri: fromUrl.pathname,
  querystring: fromUrl.searchParams.toString()
}
```

(2) I notice that the `from`s are just paths and the `to`s have the host/scheme (`https://wellcomecollection.org`) - as the host changes depending on whether it's stage or prod, and for consistency/readability, could we change this to be the same for both?

```typescript
const redirectedResponse = getRedirect(request(fromRequestObject));
const redirectLocation = redirectedResponse?.headers.location[0]?.value);
const redirectLocationUrl = new URL(redirectLocation);
redirectLocationUrl.searchParams.sort(); // as above
const toUrl = new URL(to);
toUrl.searchParams.sort();

expect(redirectedResponse?.status).toBe('301'); // was there a reason this was removed?
expect(redirectLocationUrl.pathname).toBe(toUrl.pathname);
expect(redirectLocationUrl.searchParams.toString()).toBe(toUrl.searchParams.toString());
```

Reply:
Thank you for the above, I've pushed some changes and opted to keep only the pathname in the array for readability, let me know if any of it could still be better and/or doesn't reflect what you had in mind.